app/controllers/jobseekers/job_applications/base_controller.rb

```diff
@@ -2,9 +2,7 @@ class Jobseekers::JobApplications::BaseController < Jobseekers::BaseController
   helper_method :job_application
 
   def step_process
-    Jobseekers::JobApplications::JobApplicationStepProcess.new(
-      job_application: job_application,
-    )
+    job_application.step_process
```

Comment:
think this line can look at the vacancy and create the appropriate step_process for the job application. IMHO step_process is a controller concern, and we might want other flows which are nothing to do with job application type (which I'm still unsure of, as they are totally dependent on the vacancy type (hence this seems to violate DRY principles) - looking at the vacancy we can always obtain job application 'type' (hence step process)

Reply:
Agree on this point. Flow flow-related process would be better living outside of the model.